integration-tests/smoke/vrfv2plus_test.go

```diff
@@ -1364,6 +1301,199 @@ func TestVRFV2PlusWithBHS(t *testing.T) {
+		randomWordsFulfilledEvent, err := vrfv2plus.RequestRandomnessAndWaitForFulfillment(
+			vrfv2PlusContracts.VRFV2PlusConsumer[1],
+			vrfv2PlusContracts.CoordinatorV2Plus,
+			vrfv2PlusData,
+			fundedSubIDs[0],
+			isNativeBilling,
+			configCopy.VRFv2Plus.General,
+			l,
+		)
+		require.NoError(t, err, "error requesting randomness and waiting for fulfilment")
+		require.True(t, randomWordsFulfilledEvent.Success, "RandomWordsFulfilled Event's `Success` field should be true")
+
+		// 4. wait for the request timeout (1s more) duration
+		time.Sleep(timeout + 1*time.Second)
+
+		// 5. fund sub so that node can fulfill request
+		err = vrfv2plus.FundSubscriptions(
+			env,
+			fundingLinkAmt,
+			fundingNativeAmt,
+			linkToken,
+			vrfv2PlusContracts.CoordinatorV2Plus,
+			[]*big.Int{subID},
+		)
+		require.NoError(t, err, "error funding subs after request timeout")
+
+		// 6. no fulfilment should happen since timeout+1 seconds passed in the job
+		pendingReqExists, err := vrfv2PlusContracts.CoordinatorV2Plus.PendingRequestsExist(testcontext.Get(t), subID)
+		require.NoError(t, err, "error fetching PendingRequestsExist from coordinator")
+		require.True(t, pendingReqExists, "pendingRequest must exist since subID was underfunded till request timeout")
+
+		// 7. remove job and add new job with requestTimeout = 1 hour
+		vrfNode, exists := nodesMap[vrfcommon.VRF]
+		require.True(t, exists, "VRF Node does not exist")
+		resp, err := vrfNode.CLNode.API.DeleteJob(vrfNode.Job.Data.ID)
+		require.NoError(t, err, "error deleting job after timeout")
+		require.Equal(t, resp.StatusCode, 204)
+
+		chainID := env.EVMClient.GetChainID()
+		config.VRFv2Plus.General.VRFJobRequestTimeout = ptr.Ptr(blockchain.StrDuration{Duration: time.Duration(time.Hour * 1)})
+		vrfJobSpecConfig := vrfcommon.VRFJobSpecConfig{
+			ForwardingAllowed:             *config.VRFv2Plus.General.VRFJobForwardingAllowed,
+			CoordinatorAddress:            vrfv2PlusContracts.CoordinatorV2Plus.Address(),
+			FromAddresses:                 vrfNode.TXKeyAddressStrings,
+			EVMChainID:                    chainID.String(),
+			MinIncomingConfirmations:      int(*config.VRFv2Plus.General.MinimumConfirmations),
+			PublicKey:                     vrfv2PlusData.PubKeyCompressed,
+			EstimateGasMultiplier:         *config.VRFv2Plus.General.VRFJobEstimateGasMultiplier,
+			BatchFulfillmentEnabled:       *config.VRFv2Plus.General.VRFJobBatchFulfillmentEnabled,
+			BatchFulfillmentGasMultiplier: *config.VRFv2Plus.General.VRFJobBatchFulfillmentGasMultiplier,
+			PollPeriod:                    config.VRFv2Plus.General.VRFJobPollPeriod.Duration,
+			RequestTimeout:                config.VRFv2Plus.General.VRFJobRequestTimeout.Duration,
+			SimulationBlock:               config.VRFv2Plus.General.VRFJobSimulationBlock,
+			VRFOwnerConfig:                nil,
+		}
+
+		l.Info().Msg("Creating VRFV2 Plus Job with higher timeout (1hr)")
+		job, err := vrfv2plus.CreateVRFV2PlusJob(
```

Comment:
can we have job creation and `WaitForRandomWordsFulfilledEvent` run in parallel - I just want to avoid having issues with race conditions